exec/java-exec/src/main/java/org/apache/drill/exec/expr/EvaluationVisitor.java

```diff
@@ -106,19 +163,32 @@ public HoldingContainer visitFunctionCall(FunctionCall call, ClassGenerator<?> g
     @Override
     public HoldingContainer visitBooleanOperator(BooleanOperator op,
         ClassGenerator<?> generator) throws RuntimeException {
+      HoldingContainer hc = getPrevious(op);
+      if (hc != null) {
+        return hc;
+      }
+      newScope();
```

Comment:
Is it possible to move the call of newScope() and leaveScope() to ClassGenerator.nestEvalBlock(), unnestEvalBlock()? Those two methods would essentially create / remove a new nested scope in the code, and is actually called for BooleanOperator (and/or) and IfExpression. 

Otherwise, we have to remember to call newScope() / leaveScope() everywhere the codes creates/delete a new scope, which seems to be error-prone.